src/data/nat/factorial.lean

```diff
@@ -143,4 +143,37 @@ begin
   { exact add_factorial_succ_le_factorial_add_succ i h }
 end
 
+section desc_fac
+
+/-- desc_fac n k = (n + k)! / n!, but implemented in a recursive way for calculation.
+This is closely related to `ring_theory.polynomial.pochhammer`, but much less general. -/
```

Comment:
Is there an obstruction to making this relation explicit? Please add the specialization glue.

Reply:
doing it! is it worth stripping out some of the explicit results about ℕ in `pochhammer` and making them about `desc_fac` (e.g. `factorial_mul_pochhammer'`)? cc @semorrison

Reply:
argh, there's also a little work to do in `bernstein` too... I think `iterate_derivative_at_0_aux₁` basically comes for free, though, I used it in `card_embedding` so I'll make this API more expansive